### ISSUE TITLE
[Bug report?] Remove leading/trailing spaces from title upon creation of objects

### ISSUE THREAD

**Issue by echo6ix** [stashapp/stash#3262]
**Describe (the bug?)**
The user can currently create an object title (such as a scene) with leading spaces. The leading spaces don't seem to be displayed anywhere in the UI except for the edit tab of the object. This throws off sorting of objects by title as they're getting sorted by the leading space(s).

**To Reproduce**
1. Go to `Scenes`
2. Click on 'New'
3. Create the title `         XYZ'
4. Save
5. Sort all scenes by `Title`
  - You will notice the leading spaces aren't displayed
  - You will also notice the object isn't sorted by XYZ, but rather the leading spaces

**Expected behavior**
Remove leading/trailing spaces of a title (and performer name) upon creation of an object.

**Stash Version: (from Settings -> About):**
v0.18.0-15-gce080c1b

**Additional context**
Implementing the above would fix any newly created objects going forward, but there could potentially be existing objects with leading/trailing spaces (due to accidental leading/trailing spaces or from scrapes with leading/trailing spaces.) I don't know the optimal way to implement the following, but some kind of sanitization process, perhaps when the app initializes with each new version release, to strip all leading/trailing spaces from the titles (and performer names).

**Comment by DogmaDragon** [comment]
Reproduced on https://github.com/stashapp/stash/commit/e9fa7d071e3587d393cedca735874214e06ff9d9.

**Comment by DogmaDragon** [comment]
This is an interesting case. Some people are taking advantage of leading spaces for tag names since it allows more control over the order they are displayed. 

But not that useful on scene titles or performer names.

**Comment by ALonelyJuicebox** [comment]
> This is an interesting case. Some people are taking advantage of leading spaces for tag names since it allows more control over the order they are displayed.
> 
> But not that useful on scene titles or performer names.

True, but as someone who's taking advantage of leading spaces for tag names, I wouldn't hold back a perfectly valid bug report to support what is effectively a bit of a hack. I've submitted a request for a proper implementation of Tag Groups. 🙂https://github.com/stashapp/stash/issues/3469
